priv/emqx_auth_ldap.schema

```diff
@@ -89,6 +89,70 @@ end}.
   {datatype, string}
 ]}.
 
+{mapping, "auth.ldap.custom_base_dn", "emqx_auth_ldap.custom_base_dn", [
+  {default, "${username_attr}=${user},${device_dn}"},
+  {datatype, string}
+]}.
+
+%% auth.ldap.filters.1.key = "objectClass"
+%% auth.ldap.filters.1.value = "mqttUser"
+%% auth.ldap.filters.1.op = "and"
+%% auth.ldap.filters.2.key = "uiAttr"
+%% auth.ldap.filters.2.value "someAttr"
+%% auth.ldap.filters.2.op = "or"
+%% auth.ldap.filters.3.key = "someKey"
+%% auth.ldap.filters.3.value = "someValue"
+%% The configuratation structure sent to the application:
+%%   [{"objectClass","mqttUser"},"and",{"uiAttr","someAttr"},"or",{"someKey","someAttr"}]
+%% The resulting LDAP filter would look like this:
+%% ==> "|(&(objectClass=Class)(uiAttr=someAttr)(someKey=someValue))"
+{translation, "emqx_auth_ldap.filters",
```

Comment:
Do think it would be possible to add some simple validation to `or` and `and` `ops` so it is a good error message if something else is provided? 

Is there some format requirement for the attr `key` or `values`?

Reply:
There is a validation is the source code. Any other value and there will be a function clause exception. I found that cuttlefish reporting on bad parameters is a bit unreliable. Sometimes I didn't get any indication why the node didn't start. So I'd rather leave it as is. Furthermore, this kind of configuration is cumbersome (to say the least). I would rather push for adopting the RMQ's LDAP plugin althogether.